### PR TITLE
Bookstore: use Redux in React components

### PR DESCRIPTION
**Pull request by Yazino12**
### In this Pull:

- Initialise books reducer default state with 3 books
- Import Redux `provider` and the `store` in index.js
- Connect the app to the Redux store
- Use `useSelector` to consume the state
- Import action creators for dispatch
- Add `RemoveBook` functionality and create a new file
- Add functionality for `checkStatus` and `dispatch` the corresponding action
- Deploy App

### CODE REVIEW

File: src/components/BookStore.js [hunk 1]
```diff
@@ -1,27 +1,37 @@
...
-  ]);
+  const books = useSelector((state) => state.books);
+  const dispatch = useDispatch();
+  // const [books, setBooks] = useState([
+  //   {
+  //     id: uuidv4(),
+  //     title: 'Harry Potter',
+  //     author: 'J. K. Rowling',
+  //   },
+  // ]);
```
Comment (Mihndim2020) [inline src/components/BookStore.js]: - [x] Please kindly remove `commented code` from your `code base`, as it makes your `code base` dirty and less readable. To make sure that your codebase is always clean, always remember to remove commented-codes.

File: src/components/BookStore.js [hunk 2]
```diff
@@ -1,27 +1,37 @@
...
 
   const addBook = (title, author) => {
     const newBook = {
       id: uuidv4(),
       title,
       author,
     };
-    setBooks([...books, newBook]);
+    // setBooks([...books, newBook]);
```
Comment (Mihndim2020) [inline src/components/BookStore.js]: - [x] Please kindly remove `commented code` from your `code base`, as it makes your `code base` dirty and less readable. To make sure that your codebase is always clean, always remember to remove commented-codes.

File: src/components/BookStore.js [hunk 3]
```diff
@@ -1,27 +1,37 @@
...
+    dispatch(add(newBook));
+  };
+
+  const removeBook = (bookID) => {
+    // setBooks(books.filter((book) => book.id !== bookID));
```
Comment (Mihndim2020) [inline src/components/BookStore.js]: - [x] Please kindly remove `commented code` from your `code base`, as it makes your `code base` dirty and less readable. To make sure that your codebase is always clean, always remember to remove commented-codes.

**Review by Mihndim2020 — CHANGES_REQUESTED** [review]
### Hi @Yazino12, 

### Good job so far!
### To Highlight! :clap:  :green_circle: 

:heavy_check_mark:  Good commit messages
:heavy_check_mark: Descriptive Readme file
:heavy_check_mark: Your app is running as expected.

There are some issues that you still need to work on to get this project approved, but you are almost there!
ALMOST-MINION

# Status: Requires Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/components/BookStore.js (hunk 1 above)
Comment (Yazino12) [reply src/components/BookStore.js]: Hi @Mihndim2020, thanks for the details. I will make these changes right away 😊

**Review by prolajumokeoni — APPROVED** [review]
Hi @Yazino12, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 
## Highlights
- No linter errors ✅ 
- Git flow was used correctly 🚀 
- Good design 💯 
- Good Pr title and description 💪🏿 

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification @prolajumokeoni.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**